### DISCUSSION TITLE
New platform for standing desks?

### DISCUSSION THREAD

**Discussion by abmantis** [home-assistant/architecture#958] [Entity Models]
Currently, the platform that best represents a motorized standing desk from a UX perspective is the `cover`:  has direct position control and report, and also has buttons for up/down/stop. Still, a desk is not a cover, so it may also be confusing for a user to see the desk show up as a cover.

Another suggestion was to use the `number` platform, but from a UX perspective that does not offer the user the same type of controls (missing up/down/stop).

Should HA have a dedicated `desk` platform? Should it instead have a more generic one like `elevator` (would also fit building elevators!)? Should we just use `cover`?

**Comment by Bennett-Wendorf** [comment]
It seems this is a bit of a dead topic, but I'm hoping to bring it back to life.

I recently started working on a custom integration (to eventually be posted to HACS) for control of [Uplift Desks](https://www.upliftdesk.com/) over BLE with their Bluetooth adapter. My integration can be found [here](https://github.com/Bennett-Wendorf/hass-uplift-desk).

I agree with you that there really is no great way to implement standing desks in HA at the moment. `cover` sort of works, but as you mentioned, is clunky from a user perspective as the terminology doesn't really work for desks. In addition, HA now display a notice on their developer docs for the `cover` entity stating that it shouldn't be used for devices that aren't a cover or otherwise open and close. Frankly, I agree with them that it really makes no sense. 

In my case, my plan is to implement a multi-platform device with a number entity alongside as many as 5 button entities. Uplift specifically only supports sit, stand, stop, raise, and lower, and does not support setting a specific position. I think this would fit very nicely in a specific platform, so long as it supports feature declarations like the `cover` entity does (i.e. whether the device supports setting a position, etc.).

To that end, I wholeheartedly agree that a new `desk` platform, or something more generic like `elevator` would make a ton of sense. I personally really like the idea of something a bit more generic. I don't have a great idea of what name might make sense, so that may take more research on what types of devices would fall into this category. However, something that supports the following seems appropriate to me:
- `height` - sensor/number/read-only value (This could also be a more generic distance or something to accommodate horizontal devices?)
- `is_raising` - bool/binary sensor (Could also use wording improvement
- `is_lowering` - bool/binary sensor (Could also use wording improvement
- `raised` - control for moving to preset standing height (Wording needs improvement)
- `lowered` - control for moving to preset sitting height (Wording needs improvement)
- `raise` - control for moving up a bit (push and hold would be ideal to mimic physical "up" button)
- `lower` - control for moving down a bit (push and hold would be ideal to mimic physical "down" button)
- `stop` - control to stop current action. Similar to `cover`
- `set_position` - control for setting a specific position

**Reply by pomianowski** [reply]
Something like this, but available out of the box would be great
![image](https://github.com/user-attachments/assets/b4786e2a-49f0-41db-95a2-07b010f74243)

**Reply by Bennett-Wendorf** [reply]
> Something like this, but available out of the box would be great
> 
> ![image](https://github.com/user-attachments/assets/b4786e2a-49f0-41db-95a2-07b010f74243)
> 
> 

Exactly. We'd need to figure out what controls and sensors could actually be standardized though. For example, the integration I wrote for Uplift only supports a fraction of these features currently. I think it would fall in line with other platforms as well though where basic options are included and some integrations may include others.

**Comment by abmantis** [comment]
So, coming back to this proposal which is now almost 2 years old and knowing how the architecture side of HA works a bit more than I did at the time, unfortunately I don't believe it makes sense to add a new platform for now.

AFAIK, new platforms are only added if they benefit multiple existing official integrations and currently there is only one desk integration in HA (Idasen Desk). 
Personally, I would love to have a proper desk platform with a nice UI, but I don't think it makes sense for that to be considered until we have more integrations for desks.

Hopefully some of the custom integrations will make their way into official ones ( @Bennett-Wendorf  :eyes: :wink: ).

**Comment by abmantis** [comment]
Closing this for now, as per https://github.com/home-assistant/architecture/discussions/958#discussioncomment-13033789

**Comment by zbuh** [comment]
Why just not add a new type within the same cover integration?
<img width="652" height="857" alt="image" src="https://github.com/user-attachments/assets/db72f295-7408-46f9-855e-558c6b7c2fc6" />

**Reply by Bennett-Wendorf** [reply]
As I mentioned in my first message, HomeAssistant includes a notice in the developer docs that states specifically that cover platforms should not be used for devices that don't open and close. On top of that, the terminology just doesn't really work and would be confusing to users.

**Reply by pomianowski** [reply]
> As I mentioned in my first message, HomeAssistant includes a notice in the developer docs that states specifically that cover platforms should not be used for devices that don't open and close. On top of that, the terminology just doesn't really work and would be confusing to users.

Thanks @Bennett-Wendorf  for your clarification. Since we don't want to mislead users, and we have quite a few users who have smart desks in their smart homes, why not add such an entity?

**Reply by Bennett-Wendorf** [reply]
Unfortunately that's not really my call to make. As @abmantis mentioned previously, I don't think it's likely that the HA team would go for a new platform that doesn't benefit multiple official integrations. It's definitely something I'd love to look into in the future, but for now I just don't think it makes a ton of sense until we can take care of a lot of the grunt work to implement official integrations for more desks. 

That being said, I'm still working on my Uplift Desk custom integrations when I have time. It's not stable enough for my liking to attempt to make it an official integration, but I'm still hoping to get it there one day. I think a much stronger case could be made for a desk/elevator platform once we have at least 2 official integrations representing smart desks.
